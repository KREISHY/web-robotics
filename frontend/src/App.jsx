import React from "react";
import 'bootstrap/dist/css/bootstrap.min.css';
import { Container, Nav, Navbar } from 'react-bootstrap';
import './style.css';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from './Pages/Home';
import Tests from './Pages/Tests';
import Library from './Pages/Library';
import MyTests from './Pages/MyTests';
import Profile from './Pages/Profile';
import AddTest from './Pages/AddTest';
import TestPassing from './Pages/TestPassing';
import AddExercise from './Pages/AddExercise';
import ExercisePassing from './Pages/ExercisePassing';
import usePermissions from './hooks/usePermissions';
import EditTest from './Pages/EditTest';
import ResultTest from './Pages/ResultTest';
import TestResults from './Pages/TestResultDetailed';
import FinallResult from './Pages/Components/FinallResult';
import Footer from './Footer';

function App() {
  const { isAllowed, isLoading, error } = usePermissions();

  if (isLoading) {
    // Показать экран загрузки, пока идет проверка прав
    return <div>Loading...</div>;
  }

  if (error) {
    // Показать сообщение об ошибке
    return <div>Error: {error.message}</div>;
  }

  return (
    <div className="app-container">
      <Navbar collapseOnSelect expand="md" bg="light" variant="light" className="custom-navbar">
        <Container className="center-nav">
          <Navbar.Toggle aria-controls="responsive-navbar-nav" className="custom-nav-toggle" />
          <Navbar.Collapse id="responsive-navbar-nav" style={{ position: "relative" }}>
            <Nav className="mx-auto">
              <Nav.Link href="/" className="custom-nav-link">Главная</Nav.Link>
              <Nav.Link href="/tests" className="custom-nav-link">Тесты</Nav.Link>
              {isAllowed && (
                <>
                  {/*<Nav.Link href="/library" className="custom-nav-link">Библиотека</Nav.Link>*/}
                  <Nav.Link href="/my-tests" className="custom-nav-link">Мои тесты</Nav.Link>
                  <Nav.Link href="/result-test" className="custom-nav-link">Результаты</Nav.Link>
                </>
              )}
              <Nav.Link href="/profile" className="custom-nav-link">Профиль</Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
      <div className="main-content">
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/tests" element={<Tests />} />
            {isAllowed && (
              <>
                <Route path='/my-tests/edit-test/test/:id' element={<EditTest />} />
                <Route path="/library" element={<Library />} />
                <Route path="/my-tests" element={<MyTests />} />
                <Route path="/my-tests/add-test" element={<AddTest />} />
                <Route path="/my-tests/add-exercise" element={<AddExercise />} />
                <Route path="/my-tests/test/:id" element={<TestPassing />} />
                <Route path="/result-test" element={<ResultTest />} />
                <Route path="/result-test/test/:id" element={<TestResults />} />
                <Route path="/exercise/:id" element={<ExercisePassing />} />
              </>
            )}
            <Route path="/profile/*" element={<Profile />} />
            <Route path="/public-tests/test/:id" element={<TestPassing />} />
            <Route path="/public-tests/test/result/:id" element={<FinallResult />} />
          </Routes>
        </BrowserRouter>
      </div>
      <Footer />
    </div>
  );
}

export default App;