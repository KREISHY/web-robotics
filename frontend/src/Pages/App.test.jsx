import axios from 'axios';
jest.mock('axios');
import React from "react";
import {getByRole, render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import App from "../App";

describe('App testing', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Navbar renders', () => {
        render(<App />)
        
        const homeLink = screen.getByRole('link', {name: 'Главная'});
        const testsLink = screen.getByRole('link', {name: 'Тесты'});
        const libraryLink = screen.getByRole('link', {name: 'Библиотека'});
        const myTestsLink = screen.getByRole('link', {name: 'Мои тесты'});
        const profileLink = screen.getByRole('link', {name: 'Профиль'});
        

        expect(homeLink).toBeInTheDocument();
        expect(testsLink).toBeInTheDocument();
        expect(libraryLink).toBeInTheDocument();
        expect(myTestsLink).toBeInTheDocument();
        expect(profileLink).toBeInTheDocument();
    })

    

});