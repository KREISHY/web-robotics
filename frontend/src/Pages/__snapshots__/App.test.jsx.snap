// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light"
  >
    <div
      class="custom-container container"
    >
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="custom-nav-toggle navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="modal-fullscreen navbar-nav"
        >
          <a
            class="custom-nav-link nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            Главная
          </a>
          <a
            class="custom-nav-link nav-link"
            data-rr-ui-event-key="/tests"
            href="/tests"
          >
            Тесты
          </a>
          <a
            class="custom-nav-link nav-link"
            data-rr-ui-event-key="/library"
            href="/library"
          >
            Библиотека
          </a>
          <a
            class="custom-nav-link nav-link"
            data-rr-ui-event-key="/my-tests"
            href="/my-tests"
          >
            Мои тесты
          </a>
          <a
            class="custom-nav-link nav-link"
            data-rr-ui-event-key="/profile"
            href="/profile"
          >
            Профиль
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <div
      class="color"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <img
              alt="изображения"
              src="new_img1.png"
              width="700px"
            />
          </div>
          <div
            class="col"
          >
            <h1
              class="hwhite"
            >
              Добро пожаловать в "НАЗВАНИЕ"
            </h1>
            <span
              class="white-text"
            >
              «НАЗВАНИЕ» — это инновационное веб-приложение, созданное специально для логопедов, чтобы помочь вам в работе с пациентами, страдающими дизартрией.
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <h1>
            Как начать?
          </h1>
          <ul>
            <li>
               Зарегистрируйтесь или войдите в свою учетную запись.
            </li>
            <li>
               Изучайте упражнения: Просматривайте каталог упражнений и выбирайте подходящие для ваших пациентов.
            </li>
            <li>
               Создавайте свои упражнения: Используйте нашу библиотеку материалов для создания уникальных занятий.
            </li>
            <li>
               Отслеживайте прогресс: Сохраняйте результаты и анализируйте успехи ваших пациентов.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
